daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -594,6 +594,12 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
 
       case EFromAnyTemplate(tmplId, e) =>
         SEApp(SEBuiltin(SBFromAnyTemplate(tmplId)), Array(translate(e)))
+
+      case ETyCon(tyCon) =>
+        // Ensure that the type is known.
```

Comment:
I think this check should probably be done during typechecking not in the compiler.

Reply:
Yes, it'll be checked in the typechecker. I've removed this for being redundant.